Compiler Warnings

When compiling with -Wall -Wno-unused, I get the following warnings:

```
Warnings   << swri_console:make ws_root/logs/swri_console/build.make.008.log                                                                                                                                         
ws_root/src/swri_console/src/node_list_model.cpp: In member function ‘std::string swri_console::NodeListModel::nodeName(const QModelIndex&) const’:
ws_root/src/swri_console/src/node_list_model.cpp:61:36: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
       index.row() > ordering_.size()) {
                                    ^
ws_root/src/swri_console/src/node_list_model.cpp: In member function ‘virtual QVariant swri_console::NodeListModel::data(const QModelIndex&, int) const’:
ws_root/src/swri_console/src/node_list_model.cpp:71:36: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
       index.row() > ordering_.size()) {
                                    ^
ws_root/src/swri_console/src/console_window.cpp: In member function ‘void swri_console::ConsoleWindow::nodeSelectionChanged()’:
ws_root/src/swri_console/src/console_window.cpp:226:41: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   for (size_t i = 0; i < selection.size(); i++) {
                                         ^
ws_root/src/swri_console/src/console_window.cpp:234:42: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   for (size_t i = 0; i < node_names.size(); i++) {
                                          ^
In file included from ws_root/src/swri_console/src/ros_thread.cpp:32:0:
ws_root/src/swri_console/include/swri_console/ros_thread.h: In constructor ‘swri_console::RosThread::RosThread()’:
ws_root/src/swri_console/include/swri_console/ros_thread.h:76:10: warning: ‘swri_console::RosThread::is_connected_’ will be initialized after [-Wreorder]
     bool is_connected_;
          ^
ws_root/src/swri_console/include/swri_console/ros_thread.h:75:19: warning:   ‘volatile bool swri_console::RosThread::is_running_’ [-Wreorder]
     volatile bool is_running_;
                   ^
ws_root/src/swri_console/src/ros_thread.cpp:36:1: warning:   when initialized here [-Wreorder]
 RosThread::RosThread() :
 ^
ws_root/src/swri_console/src/log_database_proxy_model.cpp: In member function ‘virtual QVariant swri_console::LogDatabaseProxyModel::data(const QModelIndex&, int) const’:
ws_root/src/swri_console/src/log_database_proxy_model.cpp:268:40: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
       index.row() >= msg_mapping_.size()) {
                                        ^
ws_root/src/swri_console/src/log_database_proxy_model.cpp: In member function ‘void swri_console::LogDatabaseProxyModel::saveTextFile(const QString&) const’:
ws_root/src/swri_console/src/log_database_proxy_model.cpp:462:40: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   for(int i = 0; i < msg_mapping_.size(); i++)
                                        ^
cd ws_root/build/swri_console; catkin build --get-env swri_console | catkin env -si  /usr/bin/make --jobserver-fds=6,7 -j; cd -
```

--- a/include/swri_console/ros_thread.h
+++ b/include/swri_console/ros_thread.h
@@ -72,8 +72,8 @@ namespace swri_console
     void startRos();
     void stopRos();
 
-    volatile bool is_running_;
     bool is_connected_;
+    volatile bool is_running_;
     ros::Subscriber rosout_sub_;
   };
 }

--- a/src/console_window.cpp
+++ b/src/console_window.cpp
@@ -223,7 +223,7 @@ void ConsoleWindow::nodeSelectionChanged()
   std::set<std::string> nodes;
   QStringList node_names;
 
-  for (size_t i = 0; i < selection.size(); i++) {
+  for (int i = 0; i < selection.size(); i++) {
     std::string name = node_list_model_->nodeName(selection[i]);
     nodes.insert(name);
     node_names.append(name.c_str());
@@ -231,7 +231,7 @@ void ConsoleWindow::nodeSelectionChanged()
 
   db_proxy_->setNodeFilter(nodes);
 
-  for (size_t i = 0; i < node_names.size(); i++) {
+  for (int i = 0; i < node_names.size(); i++) {
     node_names[i] = node_names[i].split("/", QString::SkipEmptyParts).last();
   }
     

--- a/src/log_database_proxy_model.cpp
+++ b/src/log_database_proxy_model.cpp
@@ -265,7 +265,7 @@ QVariant LogDatabaseProxyModel::data(
   }
 
   if (index.parent().isValid() &&
-      index.row() >= msg_mapping_.size()) {
+      static_cast<size_t>(index.row()) >= msg_mapping_.size()) {
     return QVariant();
   }
 
@@ -459,7 +459,7 @@ void LogDatabaseProxyModel::saveTextFile(const QString& filename) const
   QFile outFile(filename);
   outFile.open(QFile::WriteOnly);
   QTextStream outstream(&outFile);
-  for(int i = 0; i < msg_mapping_.size(); i++)
+  for(size_t i = 0; i < msg_mapping_.size(); i++)
   {
     QString line = data(index(i), Qt::DisplayRole).toString();
     outstream << line << '\n';

--- a/src/node_list_model.cpp
+++ b/src/node_list_model.cpp
@@ -58,7 +58,7 @@ int NodeListModel::rowCount(const QModelIndex &parent) const
 std::string NodeListModel::nodeName(const QModelIndex &index) const
 {
   if (index.parent().isValid() ||
-      index.row() > ordering_.size()) {
+      static_cast<size_t>(index.row()) > ordering_.size()) {
     return "";
   }
 
@@ -68,7 +68,7 @@ std::string NodeListModel::nodeName(const QModelIndex &index) const
 QVariant NodeListModel::data(const QModelIndex &index, int role) const
 {
   if (index.parent().isValid() ||
-      index.row() > ordering_.size()) {
+      static_cast<size_t>(index.row()) > ordering_.size()) {
     return QVariant();
   } 
 

